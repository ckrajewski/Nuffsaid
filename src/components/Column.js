import React from 'react';
import Message from './Message';
import ColumnTitle from './ColumnTitle';

/**
  * Column component renders each column
  * Corresponding messages are rendered for each given prioirty
*/
export default function Column(props) {
  const { messages, priority, clearMessage } = props;
  const { value, label } = priority;
  return (
    <div>
      <ColumnTitle priorityLabel={label} priorityValue={value} messageCount={messages.length} />
      {messages.map((message, index) => (
        <Message
          clearMessage={() => clearMessage(value, index)}
          priority={value}
          message={message.message}
          key={value + message.message}
        />
      ))}
    </div>
  );
}
