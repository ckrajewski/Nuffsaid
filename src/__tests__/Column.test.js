import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Column from '../components/Column';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const priority = [{value:1, label:'Error'}];
const messages =[{message:'error message'},{message:'another error message'}];
describe('Column component', () => {
  test('renders error messages', () => {
    const tree = renderer.create(<Column priority={priority} messages={messages} />);
    expect(tree).toMatchSnapshot();
  });
});