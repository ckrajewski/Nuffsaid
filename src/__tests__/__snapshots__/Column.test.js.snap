// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component renders error messages 1`] = `
<div>
  <div>
    <div>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        undefined Type undefined
      </h6>
    </div>
    <div>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Count 2
      </p>
    </div>
  </div>
  <div>
    <div
      className="MuiPaper-root makeStyles-paper-33 MuiPaper-elevation5 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-grid-36 MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root makeStyles-message-34 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            error message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clear-35 MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-32"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Clear
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiPaper-root makeStyles-paper-33 MuiPaper-elevation5 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-grid-36 MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root makeStyles-message-34 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            another error message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clear-35 MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-32"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Clear
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
